scipy/cluster/_hierarchy.pyx

```diff
@@ -169,7 +175,8 @@ cpdef void cluster_maxclust_monocrit(const double[:, :] Z, const double[:] MC, i
     if not visited:
         raise MemoryError
 
-    lower_idx = 0
+    # This index corresponds to "-INFINITY". `MC` is never evaluated at this index.
+    lower_idx = -1
```

Comment:
This makes sense from the perspective that the loop would not have been entered with `n = 2`. And I see that `MC[-1]` never happens.